src/components/PostPageDetail/index.js

```diff
@@ -85,6 +87,8 @@ const PostPageDetailIdComponent = (props) => {
   const {updateFeedContext: updateTopicContext} = usePostContextHook(CONTEXT_SOURCE.TOPIC_FEEDS);
   const {width: displayWidth} = useWindowDimensions();
 
+  const {followUnfollow} = usePostHook();
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new hook `usePostHook` is introduced here. Ensure that the `followUnfollow` function returned by this hook is properly defined and handles all edge cases, as it is used later in the code for following/unfollowing users.
```diff
+ const {followUnfollow} = usePostHook();
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->